drivers/display/uc81xx.c

```diff
@@ -227,6 +229,22 @@ static int uc81xx_set_profile(const struct device *dev,
 			return -EIO;
 		}
 
+		if (config->quirks->pon_after_softstart) {
+			/* UC8151D requires PON command after BTST for proper
+			 * power initialization
+			 */
+			LOG_DBG("Sending PON command after softstart");
+			if (uc81xx_write_cmd(dev, UC81XX_CMD_PON, NULL, 0)) {
+				return -EIO;
+			}
```

Comment:
`if (xxx) return -EIO;` does not return the correct error code from the driver like `err = xxx; if (err < 0) return err;` would do, but it's already like that everywhere else, so tuning it would be a different PR.